.concourse/product-release/pipeline.template

```diff
@@ -0,0 +1,467 @@
+    tag: release
+
+resources:
+- name: kubecf-master
+  type: git
+  source:
+    branch: master
+    uri: https://github.com/SUSE/kubecf
+
+- name: catapult
+  type: git
+  source:
+    branch: master
+    uri: https://github.com/SUSE/catapult
+
+- name: s3.kubecf-bundle
+  type: s3
+  source:
+    bucket: kubecf
+    access_key_id: ((aws-access-key))
+    secret_access_key: ((aws-secret-key))
+    region_name: us-west-2
+    regexp: kubecf-bundle-v(.*).tgz
+
+{{ $pipelineName := .Env.PIPELINE }}
+
+{{- $obtain_ekcp := `
+            # obtain kubeconfig from ekcp
+            pushd catapult
+            BACKEND=ekcp make recover
+            cp build$CLUSTER_NAME/kubeconfig kubeconfig_$CLUSTER_NAME
+` }}
+
+{{- $import_gke := `
+            # create builfolder prepared for gke
+            export BACKEND=gke
+            cat '((gke-key-json))' > $PWD/gke-key.json
+            export GKE_CRED_JSON=$PWD/gke-key.json
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $import_eks := `
+            # create builfolder prepared for eks
+            export BACKEND=eks
+            export AWS_ACCESS_KEY_ID='((aws-ci-chuller-access-key-id))'
+            export AWS_SECRET_ACCESS_KEY='((aws-ci-chuller-secret-access-key))'
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $import_aks := `
+            # create builfolder prepared for aks
+            export BACKEND=aks
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $import_caasp4 := `
+            # create builfolder prepared for caasp4os
+            export BACKEND=caasp4os
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $deploy := `
+            # deploy kubecf from chart
+            export SCF_CHART="$(readlink -f ../s3.kubecf-bundle/*.tgz)"
+            export SCF_TESTGROUP=true
+            export SCF_OPERATOR=true
+            export DOCKER_ORG=cap-staging
+            export QUIET_OUTPUT=true
+            make scf
+` }}
+
+{{- $test := `
+            # run test suites on cluster
+            # See: https://github.com/SUSE/catapult/wiki/Running-SCF-tests#kubecf
+            export KUBECF_TEST_SUITE="${TEST_SUITE:-smokes}"
+            export KUBECF_NAMESPACE="scf"
+            export SCF_TESTGROUP=true
+            export QUIET_OUTPUT=true
+            make tests-kubecf
+` }}
+
+{{- $stratos := `
+            # deploy stratos console & metrics
+            export QUIET_OUTPUT=true
+            make stratos metrics
+` }}
+
+{{- $upgrade := `
+            # upgrade kubecf
+            export SCF_CHART="$(readlink -f ../s3.kubecf-bundle/*.tgz)"
+            export SCF_OPERATOR=true
+            export QUIET_OUTPUT=true
+            make scf-chart # this takes the same chart as the deploy step for now
+            make scf-gen-config
+            make scf-upgrade
+` }}
```

Comment:
These scripts should be files under the repo, rather than embedded in the template, so they can be linted and syntax highlighted. You can render them using `gomplate`, though.